src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -112,24 +147,62 @@ def from_nwb_paths(
         -------
         An instance of DatasetConverter.
         """
-        session_converters = SessionConverter.from_nwb_paths(nwb_paths=nwb_paths)
+        try:
+            session_converters = SessionConverter.from_nwb_paths(nwb_paths=nwb_paths)
+
+            dataset_description = None
+            if additional_metadata_file_path is not None:
+                dataset_description = DatasetDescription.from_file_path(file_path=additional_metadata_file_path)
+
+            session_messages = [
+                message
+                for session_converter in session_converters
+                for message in session_converter.messages
+                if session_converter.messages is not None
+            ]
 
-        dataset_description = None
-        if additional_metadata_file_path is not None:
-            dataset_description = DatasetDescription.from_file_path(file_path=additional_metadata_file_path)
+            dataset_converter = cls(session_converters=session_converters, dataset_description=dataset_description)
+            dataset_converter._internal_messages = session_messages
+            return dataset_converter
+        except:  # noqa
```

Comment:
```suggestion
        except Exception:  # noqa
```

The bare `except` would catch `SystemExit` and `KeyboardInterrupt` as well.